core/reader/replicate_channel_manager.go

```diff
@@ -174,71 +178,96 @@ func (r *replicateChannelManager) StartReadCollection(ctx context.Context, info
 
 	sourceDBInfo := r.metaOp.GetDatabaseInfoForCollection(ctx, info.ID)
 
+	var shouldSendAPIEvent bool
 	var err error
-	retryErr := retry.Do(ctx, func() error {
-		_, err = r.targetClient.GetCollectionInfo(ctx, info.Schema.GetName(), sourceDBInfo.Name)
-		if err != nil &&
-			!IsCollectionNotFoundError(err) && !IsDatabaseNotFoundError(err) {
-			return err
-		}
-		r.collectionLock.RLock()
-		_, ok := r.replicateCollections[info.ID]
-		r.collectionLock.RUnlock()
-		if ok {
-			return errors.Newf("the collection has been replicated, wait it [collection name: %s] to drop...", info.Schema.Name)
+	if r.downstream == "milvus" {
```

Comment:
I think this change will make this part of the code extremely difficult to understand and unreadable, because the current approach is equivalent to forcibly mixing multiple cases in one method. I prefer to make a unified function of the previous milvus processing logic.